types/esquery/index.d.ts

```diff
@@ -14,19 +15,35 @@ export = query;
 declare function query(ast: Node, selector: string): Node[];
 
 declare namespace query {
+    interface ESQueryOptions {
+        nodeTypeKey?: string;
+        visitorKeys?: { [nodeType: string]: readonly string[] };
+        fallback?: (node: Node) => string[];
+        matchClass?: (className: string, node: Node, ancestry: Node[]) => boolean;
+    }
+
     /** Parse a selector and return its AST. */
     function parse(selector: string): Selector;
     /** From a JS AST and a selector AST, collect all JS AST nodes that match the selector. */
-    function match(ast: Node, selector: Selector): Node[];
+    function match(ast: Node, selector: Selector, options?: ESQueryOptions): Node[];
     /** Given a `node` and its ancestors, determine if `node` is matched by `selector`. */
-    function matches(node: Node, selector: Selector, ancestry: Node[]): boolean;
+    function matches(node: Node, selector: Selector, ancestry?: Node[] | null, options?: ESQueryOptions): boolean;
     /** Query the code AST using the selector string. */
-    function query(ast: Node, selector: string): Node[];
+    function query(ast: Node, selector: string, options?: ESQueryOptions): Node[];
+
+    /** From a JS AST and a selector AST, collect all JS AST nodes that match the selector. */
+    function traverse(
+        ast: Node,
+        selector: string,
+        visitor: (node: Node, parent: Node, ancestry: Node[]) => void,
+        options?: ESQueryOptions,
+    ): Node[];
```

Comment:
The JSDoc for `traverse` say it returns a `Node[]`, but the [actual source code](https://github.com/estools/esquery/blob/7d1f6691d8f76a85a84214ce210427d6b055b196/esquery.js#L540-L579) shows that it returns `void`.

```suggestion
    ): void
```